src/main/java/com/google/devtools/build/lib/rules/java/JavaToolchain.java

```diff
@@ -160,6 +161,17 @@ public ConfiguredTarget create(RuleContext ruleContext)
 
     JavaRuntimeInfo javaRuntime = JavaRuntimeInfo.from(ruleContext, "java_runtime");
 
+    if (javaRuntime.version() != 0 && javaRuntime.version() < bootclasspath.version()) {
```

Comment:
I wondered about using `OptionalInt` in the Java code here, but I don't have a strong opinion. If starlark is using `0`, it might be clearer to be consistent